sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -662,8 +693,13 @@ private ConnectionOptions getConnectionOptions() {
             ? verifyMode
             : SslDomain.VerifyMode.VERIFY_PEER_NAME;
 
+        final String hostname = customEndpointAddress != null ? customEndpointAddress.getHost() : null;
+        final Integer port = customEndpointAddress != null && customEndpointAddress.getPort() != -1
+            ? customEndpointAddress.getPort()
+            : null;
```

Comment:
Should we just turn this into a single `if/else`

Reply:
There is another `&& customEndpointAddress().getport()` check here, that's why I didn't.